### ISSUE TITLE
Load TOML, not GeoPackage, in QGIS plugin

### ISSUE THREAD

**Issue by visr** [Deltares/Ribasim#629]
Right now QGIS loads a model from a GeoPackage, but really we should load the TOML and find the GeoPackage from there. When we do this, we can directly read the TOML, check if the output exists, and if it does, associate the node and edge output automatically on loading the model. This saves quite some clicks when loading a model for visualization.

Originally mentioned in, and needed for, #318.

**Comment by evetion** [comment]
Needs refinement. There's the possibility that we might use Ribasim Python in the QGIS plugin, which would impact this issue.

**Comment by deltamarnix** [comment]
The first part of this ticket was implemented. toml files are loaded instead of gpkg files. This was pushed to the main branch.

@visr For connecting the results I am missing some information. Currently there are 2 buttons in the Results tab. 

![Image](https://github.com/Deltares/Ribasim/assets/150045289/d12b07fb-78e2-43d3-8624-b3fd6a923c1d)

The toml file has multiple node and edge results, and I am not sure how we can associate multiple result files onto a single layer "Node" or "Edge". The arrow files that are written are: flow, basin, allocation, and control.

**Comment by visr** [comment]
Right, only `basin.arrow` and `flow.arrow` files are currently supported in QGIS. Associate Node Results means `basin.arrow`, and Edge results means 'flow.arrow. See also https://deltares.github.io/Ribasim/qgis/#sec-results.

For control results there is https://github.com/Deltares/Ribasim/issues/342, for allocation results there is no issue yet, that was only just added.

Related is also an item of https://github.com/Deltares/Ribasim/issues/638, where we suggested always writing the paths in `[results]` so we don't have to know the default location of the output in the QGIS plugin.

**Comment by deltamarnix** [comment]
While working on this issue I figured that it is very hard for us to do development on this plugin. Therefore I have started investigating how to setup the development environment.

So far I have been installing QGIS via conda forge so that I have the correct dependencies installed. And I have created a pyproject.toml inside of ribasim_qgis.

To ensure that all variables are set in python and found by vs code, it's easiest to run `pixi run code .` on the command line in order to open vscode with the right parameters. This resolves all the pylance onfound imports.

I am still having some small issues with mypy, and I saw that the plugin path of QGIS is pointing to .pixi\env\Library\plugins. This is not the folder where ribasim_qgis is installed. Maybe we can add that to the installation script? Or we have to set a different plugin path.
